tls/s2n_client_hello.c

```diff
@@ -863,7 +864,7 @@ int s2n_client_hello_get_parsed_extension(s2n_tls_extension_type extension_type,
     POSIX_GUARD(s2n_extension_supported_iana_value_to_id(extension_type, &extension_type_id));
 
     s2n_parsed_extension *found_parsed_extension = &parsed_extension_list->parsed_extensions[extension_type_id];
-    POSIX_ENSURE_REF(found_parsed_extension->extension.data);
+    POSIX_ENSURE(found_parsed_extension->extension.data, S2N_ERR_EXTENSION_NOT_RECEIVED);
```

Comment:
Just to double check-- this doesn't fail for a zero-length extension, right? We still set data, just with a zero length?

Reply:
Correct - although the reason seems a bit weird. This line is checking that the data blob on the extension stuffer isn't null. After receiving a zero-length extension, we do a raw read with a zero length, which still returns a valid pointer:
https://github.com/aws/s2n-tls/blob/9014236e7ba83869251298341264ccf1a37cd9f6/tls/extensions/s2n_extension_list.c#L134
This is then used to init the blob, so it's a zero length blob with non-nulll data.

I didn't see a test for this so I added one.